Installing-and-Configuring/OS/index.md

```diff
@@ -0,0 +1,136 @@
+
+The attacker should neither have access to the LUKS Disk Recovery Key (LUKS keyslot 0) nor its associated passphrase, because it was setuped at install time and should only be typed from the user when renewewing/changing the Disk Unlock Key passphrase when setting a new boot default after Heads detects changed core /boot components and prompts user to detach sign /boot digest (sha256sum).
+
+Even if the attacker clones the drive, he would still not have access to the decrypted data, unless he eavesdropped the Disk Recovery Key passphrase which was prompted when defining a new boot default, which should be done right after having upgraded QubesOS dom0, from a safe place where eavesdropping is highly improbable.
+
+To mitigate recovery shell attacks, entering the recovery shell will modify the PCR and measurements. Also changes to kernel or initrd files on /boot will modify the measurements.  This will be detected as tampering.
+
+See also [TPM vs security dongle](#tpm-vs-security-dongle)
+
+#### Disable TPM Disk Unlock passphrase/key
+
+This feature may be turned off during configuration of heads.  The following will instruct heads not to ask for a disk unlock passphrase and to not store a disk unlock key in the TPM.  
+
+  CONFIG_TPM_NO_LUKS_DISK_UNLOCK=y
+
+### Using Heads Disk Recovery Key 
+
+In the Disk Recovery Key scenario, Heads doesn't attempt to release the Disk Unlock Key. It simply relies on OS mechanisms to directly prompt the user for the Disk Recovery Key passphrase at standard OS prompt.  
+
+This model is vulnerable to brute force attacks.
+
+### Using an OS managed LUKS passphrase (not heads)
+
+In this model heads is not involved in processes related to the decryption of data on disk. The OS manages keys and access to the LUKS containers by asking the user for a passphrase on boot.  
+
+This model is vulnerable to brute force attacks.
+
+### Security Dongle with LUKS root (not heads)
+
+In this model heads is not responsible for any of the decryption of data on disk.  You may use a security dongle (aka smartcard aka security token) to manage the passphrase for the root LUKS partition.  Like the TPM, the security dongle enforces limits which prevent brute force attacks.
+
+In contrast to [using TPM for decryption](#using-tpm-for-decryption-of-drives)--where a disk unlock passphrase is stored in the TPM--in this model the TPM is not used for LUKS decryption.  A LUKS disk unlock key/passphrase is encrypted and stored in a 'public' location such as /boot.  After heads boots the kernel a process which uses the security dongle decrypts the LUKS root partition.
+
+See also [TPM vs security dongle](#tpm-vs-security-dongle)
+
+TPM vs Security Dongle
+----
+
+Which one you choose depends on hardware you have and threat models you are working with.  Your system may not have a TPM or you may not have a smartcard.  The core functionality is similar in that they both store private keys and allow for cryptographic operations.
```

Comment:
This is tricky to explain. 

Heads doesn't "require" a TPM while strongly suggested. But it requires a USB Security dongle enforcing GPG operations for /boot digest detach signing. As you know, key generations (generating keypairs and exporting public key to USB drive, importing public key inside of the ROM) or importing USB Security dongle private key's matching public key is required to sign /boot content.

So the statement here saying "you might not have a smartcard" is a bit contradicting reality of Heads stated prerequisites. 

What is true is that a USB Security dongle doesn't require to support HOTP for remote attestation, while a USB Security dongle enforcing GPG is required https://osresearch.net/Prerequisites#usb-security-dongles-aka-security-token-aka-smartcard